gems/aws-sdk-core/lib/aws-sdk-core/middleware/checksums.rb

```diff
@@ -57,13 +57,20 @@ def call(input, context)
 
         output = @app.call(input, context)
         @response_validation_body&.validate!
+        if context.metadata[:http_checksum][:validated]
+          output.metadata[:http_checksum] = {
+            validated: context.metadata[:http_checksum][:validated]
+          }
```

Comment:
I think I had made a change for output metadata to always reference context metadata, or at least I did so for request id. I think it makes sense here, too - context metadata should just be set to output metadata, then you don't need to check for this and assign it there. If it's on the context, it would automatically be on output.

Reply:
The request_id just does:
```
        output.metadata[:request_id] = context.response.headers['x-amzn-RequestId']
```

I don't want the user facing output metadata to reference the internal context here as there is additional information (such as skip_on_suffix) that are internal details that I don't want leaking out.